src/ring_theory/polynomial/special_coefficients.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.erase_lead
+import data.polynomial.degree.lemmas
+
+/-! # Computing special coefficients of polynomials
+
+This file contains computations of certain, hopefully meaningful, coefficients of polynomials.
+
+For instance, there is a computation of the coefficients "just before" the `leading_coeff`.
+
+Let `R` be a (semi)ring.
+We introduce the "previous-to-last" coefficient `ptl`, taking four natural numbers `a b c d : ℕ`
+and two polynomials `f g : R[X]` as inputs.  We define
+```lean
+ptl a b c d f g = f.coeff a * g.coeff b + f.coeff c * g.coeff d.
+```
+This is intended simply as a helper definition to prove results without being constantly required
+to prove that some natural number is positive, that some coefficient is non-zero, that some
+polynomial has precisely some degree...  We develop enough API for `ptl` to essentially prove that
+it is linear in either one of its polynomial inputs. -/
+
+open_locale polynomial
+namespace polynomial
+
+open polynomial nat
+open_locale polynomial
+
+variables {R : Type*} [semiring R] {r : R} {f g h : R[X]} {a b c d : ℕ}
+
+/-- `ptl (a : ℕ) (b : ℕ) (c : ℕ) (d : ℕ) (f : R[X]) (g : R[X]) = f_a * g_b + f_c * g_d`,
+where `p_i` is the `i`th coefficient of the polynomial `p`.
+`ptl` is an auxilliary definition whose main purpose is to prove
+lemma `coeff_mul_nat_degree_add_sub_one`.  The intended application is to compute
+`(f * g)_(deg f + deg g - 1)`, the coefficient just before the top one of `f * g`.  -/
+def ptl (a b c d : ℕ) (f g : R[X]) : R :=
```

Comment:
This docstring doesn't tell me why you used the name `ptl`!

Reply:
Ah, I never wrote it explicitly: Previous-To-Last.  As I imagine that this is not going to be particularly useful outside of this computation, I thought of making it slightly obscure, instead of using `private`.  I really do not like `private`!